### PR TITLE
fix(ci): Trigger mdn/bcd-util API deployment after release

### PR DESCRIPTION
**Pull request by OnkarRuikar**
- This is the other half of https://github.com/mdn/bcd-utils/pull/24 .

This PR creates a workflow that will trigger the API deployment workflow in `mdn/bcd-util` repo.

Note: this will need a special `secrets.DISPATCH_PAT` token to be created.

### CODE REVIEW

File: .github/workflows/update_bcd-utils_api.yml
Comment (bsmth) [inline .github/workflows/update_bcd-utils_api.yml:21]: Do we need anything in the payload?

**Review by bsmth — APPROVED** [review]
The PAT is created under `DISPATCH_PAT` with the requested permissions. No objections to merging so leaving a +1 

Thanks!

File: .github/workflows/update_bcd-utils_api.yml
Comment (OnkarRuikar) [reply .github/workflows/update_bcd-utils_api.yml:21]: The current implementation of the workflows [doesn't use](https://github.com/mdn/bcd-utils/blob/3a227173bf29ed8fd073f65ed8a482faf59e8b3c/.github/workflows/deploy-prod-api.yml#L6) the `workflow_trigger`s `notes` (optional) input; it's redundant. There is no point in sending payload that is not going to be utilized.

**Comment by OnkarRuikar** [comment]
I've added a manual execution option (`workflow_dispatch`). This way we won't have to wait for BCD release for:
- testing
- unknown circumstances. Won't have to give access to `bcd-utils` repo. Maintainers of BCD repo can simply trigger this manually.

**Review by bsmth — APPROVED** [review]

**Comment by bsmth** [comment]
This one is also safe to merge, thanks Onkar

**Pull request merged**
